ouroboros-network/test/CDDL.hs

```diff
@@ -3,134 +3,215 @@
 where
 
 import System.Exit (ExitCode(..))
 import System.Process.ByteString.Lazy
-import Control.Monad
 import Control.Exception.Base (throw)
-import qualified Data.ByteString.Lazy as BS
+import Control.Monad
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString.Internal as BSI
 import Data.ByteString.Lazy (ByteString)
-import Data.ByteString.Lazy.Char8 as Char8 (putStrLn, unpack, lines)
+import Data.ByteString.Lazy.Char8 as Char8 (lines, putStrLn, unpack)
 import Data.List (intercalate)
-import Data.Word (Word32)
+
+import Test.QuickCheck
+import Test.Tasty (defaultMain, TestTree, testGroup, adjustOption)
+import Test.Tasty.QuickCheck (testProperty, QuickCheckMaxSize(..))
+
 import qualified Codec.Serialise.Class as Serialise
-import Codec.CBOR.Decoding (decodeWord, decodeListLenOf, decodeBytes)
-import qualified Codec.CBOR.Decoding as CBOR
-import qualified Codec.CBOR.Term as CBOR (Term)
+import Codec.CBOR.Decoding as CBOR (Decoder, decodeBytes ,decodeListLenOf, decodeWord)
+import Codec.CBOR.Encoding (encodeBytes, encodeListLen, encodeWord)
+import Codec.CBOR.Term as CBOR
 import Codec.CBOR.Read
+import Codec.CBOR.Write (toLazyByteString)
+
+import Ouroboros.Network.Block (BlockNo, HeaderHash, Point)
+import Ouroboros.Network.Testing.ConcreteBlock (BlockHeader (..), Block)
 
-import Ouroboros.Network.Block (Point)
 import Ouroboros.Network.Protocol.ChainSync.Type as CS
 import Ouroboros.Network.Protocol.ChainSync.Codec (codecChainSync)
-import Network.TypedProtocol.ReqResp.Type as ReqResp
-import Network.TypedProtocol.ReqResp.Codec.Cbor (codecReqResp)
-import Network.TypedProtocol.PingPong.Type as PingPong
-import Network.TypedProtocol.PingPong.Codec.Cbor (codecPingPong)
+import Ouroboros.Network.Protocol.ChainSync.Test ()
 import Ouroboros.Network.Protocol.BlockFetch.Codec (codecBlockFetch)
+import Ouroboros.Network.Protocol.BlockFetch.Test ()
 import Ouroboros.Network.Protocol.BlockFetch.Type as BlockFetch
 import Ouroboros.Network.Protocol.Handshake.Codec (codecHandshake)
 import Ouroboros.Network.Protocol.Handshake.Type as Handshake
-import Ouroboros.Network.Block (HeaderHash)
+import Ouroboros.Network.Protocol.Handshake.Test (VersionNumber)
+import Ouroboros.Network.Protocol.TxSubmission.Codec (codecTxSubmission)
+import Ouroboros.Network.Protocol.TxSubmission.Type as TxSubmission
+import Ouroboros.Network.Protocol.TxSubmission.Test (TxId, Tx)
 
 import Network.TypedProtocol.Codec
+import Network.TypedProtocol.ReqResp.Codec.Cbor (codecReqResp)
+import Network.TypedProtocol.ReqResp.Type as ReqResp
+import Network.TypedProtocol.PingPong.Type as PingPong
+import Network.TypedProtocol.PingPong.Codec.Cbor (codecPingPong)
 
--- the concrete types used for the test
-type CS = ChainSync DummyBytes (Point DummyBytes)
-type RR = ReqResp DummyBytes DummyBytes
-type BF = BlockFetch DummyBytes
-type HS = Handshake Word32 CBOR.Term
+-- These files must be in place:
+specFile :: FilePath
+specFile = "test/messages.cddl"
```

Comment:
I don't like this hardcoded relative path. It means that you can only run the test suite if you're in the `ouroboros-network/ouroboros-network/` directory.
How about embedding the file into the executable at compile time? For example using the approach taken in https://hsyl20.fr/home/posts/2019-01-15-fast-file-embedding-with-ghc.html

Reply:
I looked at the source and I'd rather leave it as it is.
It seems to be easy to fix, if we often need to call the tests from a different path
At the moment I use ghci and the paths just work for my workflow.